### ISSUE TITLE
pion ignoring the setting of transport-wide-cc RTP extension

### ISSUE THREAD

**Issue by atoppi** [pion/webrtc#2373]
### Your environment.
  - Version: *v3.1.50*

### What did you do?
I started a send-only connection from pion to Janus sfu, with twcc enabled through `RegisterDefaultInterceptors` helper.
Rid based simulcasting is being used here, but I'm not sure if it has something to do with the issue.

<details><summary>pion offer</summary>

```
v=0
o=- 8402829463526960380 1671020807 IN IP4 0.0.0.0
s=pion
t=0 0
a=fingerprint:sha-256 28:72:FC:A2:9A:28:74:1A:87:E0:5A:23:62:9A:A2:8C:D4:2E:19:FA:4C:CF:7C:CB:DA:E2:E2:72:71:F7:2D:C5
a=extmap-allow-mixed
a=group:BUNDLE 0 1
m=audio 9 UDP/TLS/RTP/SAVPF 111
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:0
a=ice-ufrag:NgGWFlvhQPGcszMk
a=ice-pwd:JGhLFDwvIrOXStQGJlCcASIDGpPWlswX
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:111 opus/48000/2
a=rtcp-fb:111 transport-cc 
a=extmap:1 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=extmap:2 urn:ietf:params:rtp-hdrext:sdes:mid
a=ssrc:3094694861 cname:pion
a=ssrc:3094694861 msid:pion audio
a=ssrc:3094694861 mslabel:pion
a=ssrc:3094694861 label:audio
a=msid:pion audio
a=sendonly
m=video 9 UDP/TLS/RTP/SAVPF 100
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:1
a=ice-ufrag:NgGWFlvhQPGcszMk
a=ice-pwd:JGhLFDwvIrOXStQGJlCcASIDGpPWlswX
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:100 VP8/90000
a=rtcp-fb:100 goog-remb 
a=rtcp-fb:100 ccm fir
a=rtcp-fb:100 nack 
a=rtcp-fb:100 nack pli
a=rtcp-fb:100 transport-cc 
a=extmap:1 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=extmap:2 urn:ietf:params:rtp-hdrext:sdes:mid
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:rtp-stream-id
a=ssrc:212332876 cname:pion
a=ssrc:212332876 msid:pion video
a=ssrc:212332876 mslabel:pion
a=ssrc:212332876 label:video
a=msid:pion video
a=ssrc:82646558 cname:pion
a=ssrc:82646558 msid:pion video
a=ssrc:82646558 mslabel:pion
a=ssrc:82646558 label:video
a=msid:pion video
a=ssrc:1611069741 cname:pion
a=ssrc:1611069741 msid:pion video
a=ssrc:1611069741 mslabel:pion
a=ssrc:1611069741 label:video
a=msid:pion video
a=rid:h send
a=rid:m send
a=rid:l send
a=simulcast:send h;m;l
a=sendonly
```
</details>

<details><summary>janus answer</summary>

```
v=0
o=- 3199915716562216906 1671023906 IN IP4 192.168.1.21
s=VideoRoom 100
t=0 0
a=group:BUNDLE 0 1
a=extmap-allow-mixed
a=msid-semantic: WMS janus
m=audio 9 UDP/TLS/RTP/SAVPF 111
c=IN IP4 192.168.1.21
a=recvonly
a=mid:0
a=rtcp-mux
a=ice-ufrag:SO1i
a=ice-pwd:OE47k9bOUGZTnmAcQmgdGo
a=ice-options:trickle
a=fingerprint:sha-256 51:38:21:2A:25:7B:94:D1:FC:1E:76:32:CE:05:3C:65:29:F6:F0:7B:24:B3:61:4E:AD:98:7E:43:42:6D:85:AA
a=setup:active
a=rtpmap:111 opus/48000/2
a=extmap:2 urn:ietf:params:rtp-hdrext:sdes:mid
a=extmap:1 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=msid:janus janusa0
a=ssrc:37318064 cname:janus
a=ssrc:37318064 msid:janus janusa0
a=ssrc:37318064 mslabel:janus
a=ssrc:37318064 label:janusa0
a=candidate:1 1 udp 2015363327 192.168.1.21 41107 typ host
a=end-of-candidates
m=video 9 UDP/TLS/RTP/SAVPF 100
c=IN IP4 192.168.1.21
a=recvonly
a=mid:1
a=rtcp-mux
a=ice-ufrag:SO1i
a=ice-pwd:OE47k9bOUGZTnmAcQmgdGo
a=ice-options:trickle
a=fingerprint:sha-256 51:38:21:2A:25:7B:94:D1:FC:1E:76:32:CE:05:3C:65:29:F6:F0:7B:24:B3:61:4E:AD:98:7E:43:42:6D:85:AA
a=setup:active
a=rtpmap:100 VP8/90000
a=rtcp-fb:100 ccm fir
a=rtcp-fb:100 nack
a=rtcp-fb:100 nack pli
a=rtcp-fb:100 goog-remb
a=rtcp-fb:100 transport-cc
a=extmap:1 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=extmap:2 urn:ietf:params:rtp-hdrext:sdes:mid
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:rtp-stream-id
a=msid:janus janusv0
a=ssrc:3444058978 cname:janus
a=ssrc:3444058978 msid:janus janusv0
a=ssrc:3444058978 mslabel:janus
a=ssrc:3444058978 label:janusv0
a=rid:h recv
a=rid:m recv
a=rid:l recv
a=simulcast:recv h;m;l
a=candidate:1 1 udp 2015363327 192.168.1.21 41107 typ host
a=end-of-candidates
```
</details>

### What did you expect?
Since twcc is being correctly negotiated, I'd expect pion setting the twcc RTP extension in the header of sent packets.

### What happened?
pion does not honor the RTP extension, as verified through a pcap.
The only used extensions are id 2 (mid) and id 3 (rid).

![Screenshot from 2022-12-14 15-41-58](https://user-images.githubusercontent.com/3405061/207626025-38e58fa2-b47f-4cb2-847d-5dafa99a2d78.png)

Basically this means that twcc feedbacks are not computable for the peer.

**Comment by langhuihui** [comment]
how to set
